src/main/java/pl/edu/agh/firecell/model/material/ConductionCoefficientProvider.java

```diff
@@ -0,0 +1,38 @@
+package pl.edu.agh.firecell.model.material;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+public class ConductionCoefficientProvider {
+
+    private final HashMap<Pair, Double> coeTable;
+
+    public ConductionCoefficientProvider() {
+        coeTable = new HashMap<>();
+    //==================================================================================================================
+        coeTable.put(new Pair(Material.AIR.ordinal(), Material.AIR.ordinal()),                              /**/ 0.008);
+        coeTable.put(new Pair(Material.AIR.ordinal(), Material.WOOD.ordinal()),                             /**/ 0.02);
+        coeTable.put(new Pair(Material.AIR.ordinal(), Material.CELLULAR_CONCRETE.ordinal()),                /**/ 0.015);
+    //==================================================================================================================
+        coeTable.put(new Pair(Material.WOOD.ordinal(), Material.WOOD.ordinal()),                            /**/ 0.2);
+        coeTable.put(new Pair(Material.WOOD.ordinal(), Material.CELLULAR_CONCRETE.ordinal()),               /**/ 0.15);
+    //==================================================================================================================
+        coeTable.put(new Pair(Material.CELLULAR_CONCRETE.ordinal(), Material.CELLULAR_CONCRETE.ordinal()),  /**/ 0.1);
+    //==================================================================================================================
+    }
+
+    public Optional<Double> getConductionCoe(Material m1, Material m2) {
+        return getConductionCoe(m1.ordinal(), m2.ordinal());
+    }
+
+    public Optional<Double> getConductionCoe(int m1, int m2) {
```

Comment:
Are we even using it? it is not a common practise to manipulate enums using their ordinals like that

Reply:
After a while I see why you put it there :P
If you delete passing the pair by integers, yoo won't even need the optional, and the map itself will have proper values because of autoi-validation, that I mentioned earlier. Since the map will always give you proper result, since you used enums. Get totally rid of ordinals and optionals.